features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/validations_api_router.jag

```diff
@@ -1,96 +1,143 @@
+                                                           queryParams);
+            if (result.status != mobileValidatorModule.ValidationStatus.VALID) {
+                response.status = 422; // Unprocessable Entity (WebDAV; RFC 4918)
+                response.contentType = "application/json";
+                var content = {status: "Validation Error", message: result.message};
+                response.content = stringify(content);
+                return;
+            }
+            response.status = 200;
+            response.content = stringify({status: "Success"});
+        }
+    });
 
-    }
-    catch (e) {
-        log.debug('An asset of type: ' + shortName + ' could not be created.The following exception was thrown: ' + e);
-        response.sendError(500, 'An asset of type: ' + shortName + ' could not be created.Please check the server logs.');
-    }
+    routeManager.register('GET', 'publisher', '/publisher/api/validate/{appType}/form/name/{formName}', function (context) {
+
+        var appType = context.params.appType;
+        var formName = context.params.formName;
+
+        if (appType == 'mobileapp') {
+            var queryParams = request.getAllParameters();
+            var partialValidation = false;
+            if (queryParams['partial']) {
+                partialValidation = Boolean(queryParams['partial']);
+            }
+            var mobileValidatorModule = require('/extensions/mobileapp/validations.js');
+            var mobileValidator = new mobileValidatorModule.Validator(null);
+            var result = mobileValidator.validateForm(formName, queryParams, partialValidation);
+
+            if (result.status != mobileValidatorModule.ValidationStatus.VALID) {
+                response.status = 422; // Unprocessable Entity (WebDAV; RFC 4918)
+                response.contentType = "application/json";
+                var content = {
+                    status: "Validation Error",
+                    validationResults: result.validationResults
+                };
+                response.content = stringify(content);
+                return;
+            }
+            response.status = 200;
+            response.content = stringify({status: "Success"});
+        }
+    });
 
-});
-    
     routeManager.register('POST', 'publisher', '/publisher/api/validations/assets/webapp/overview_context', function (context) {
         var contextVal = context.post.overview_context;
 
         try {
-
             var oldContextVal = '';
             var isContextExists = isContextExist(contextVal, oldContextVal);
             print(isContextExists);
-
-        }
-        catch (e) {
-            log.debug('An asset of type: ' + shortName + ' could not be created.The following exception was thrown: ' + e);
-            response.sendError(500, 'An asset of type: ' + shortName + ' could not be created.Please check the server logs.');
+        } catch (e) {
```

Comment:
fixed error message in commit b613c0c77f021a82c6ca7d0505b64ce41013e6f5 of PR #241